### PR TITLE
Improve AdminSlot plugin

### PR DESCRIPTION
**Pull request by Arkshine**
Originally initiated by @wopox1337 but took over because incomplete and it needed to be to split in multiple commits.

Notable changes:
* Cvars behavior. This is not really a bug, but before if you play with the cvars  `sv_visiblemaxplayers` is not adjusted (or not immediately). E.g. both cvars is set, if you don't hide the slots anymore,  `sv_visiblemaxplayers` is not updated.  Now, when a cvar change is detected and if required, `sv_visiblemaxplayers` update is immediate. This removes the ugly task as well.
* When `sv_visiblemaxplayers` needs to be reset, this will use -1 (default value), and not max clients.
* The cvars have now a min/max bounds. This will reduce the chance of abuse.
* If `amx_reservation` or `amx_hideslots` is set to 0, and if `sv_visiblemaxplayers` is set with a positive value, then it will be reset to -1.
* Added description to cvar. Same as the command, ML should be used for cvar. Ideally, a new param should exist, but it's fine to use `LANG_SERVER` for now.

### CODE REVIEW

File: plugins/adminslots.sma [hunk 1]
```diff
@@ -14,64 +14,97 @@
...
+
+new CvarHandleMaxVisiblePlayers;
 
 public plugin_init()
 {
-	register_plugin("Slots Reservation", AMXX_VERSION_STR, "AMXX Dev Team")
-	register_dictionary("adminslots.txt")
-	register_dictionary("common.txt")
-	g_ResPtr = register_cvar("amx_reservation", "0", FCVAR_PROTECTED)
-	g_HidePtr = register_cvar("amx_hideslots", "0")
```
Comment (OciXCrom) [inline plugins/adminslots.sma]: `bind_pcvar_num`?

File: plugins/lang/adminslots.txt [hunk 2]
```diff
@@ -1,5 +1,7 @@
 [en]
 DROPPED_RES = Dropped due to slot reservation
+CVAR_RESERVATION = Amount of slots to reserve
+CVAR_HIDESLOTS = If you set this to 1, you can hide slots on your server.^nIf server "full" of public slots and slots hidden, you must manually connect with connect console command.
```
Comment (OciXCrom) [inline plugins/lang/adminslots.txt]: > If server "full" of public slots and slots hidden, you must manually connect with connect console command.

=>

> If the server's public and hidden slots are full, you must manually connect with the "connect" console command.

File: plugins/adminslots.sma (hunk 1 above)
Comment (Arkshine) [reply plugins/adminslots.sma]: Cvars are hooked in another commit.

File: plugins/lang/adminslots.txt (hunk 2 above)
Comment (Arkshine) [reply plugins/lang/adminslots.txt]: I used the original text, but yes, I can change it.

**Comment by wopox1337** [comment]
The next step should be to add an API for the plugin

**Comment by OciXCrom** [comment]
Maybe also add a cvar to change the slot reservation flag?

File: plugins/lang/adminslots.txt [hunk 3]
```diff
@@ -1,5 +1,7 @@
...
 
 [de]
 DROPPED_RES = Sorry, dieser Slot ist reserviert
```
Comment (Daniele386) [inline plugins/lang/adminslots.txt]:
```suggestion
DROPPED_RES = Sorry, dieser Slot ist reserviert
CVAR_RESERVATION = Anzahl der zu reservierenden Slots
CVAR_HIDESLOTS = Wenn diese CVAR auf 1 gesetzt wird, können Slots auf Ihrem Server ausgeblendet werden.^nWenn die öffentlichen Slots auf dem Server "voll" sind und weitere freie Slots ausgeblendet sind, müssen Sie die Verbindung manuell über den Konsolenbefehl "connect" herstellen.
```

**--> updated german translation**

**Comment by wopox1337** [comment]
need to be merged @Arkshine

**Comment by wopox1337** [comment]
bump

**Comment by wopox1337** [comment]
@Arkshine also merge this

**Comment by Arkshine** [comment]
Ya.  @wopox1337 I remember testing it extensively (and actually did you too?), but don't hesitate to play with it if you see any issues.

**Pull request merged**